platform/web/detect.py

```diff
@@ -118,6 +118,11 @@ def configure(env: "Environment"):
         else:
             env.Append(CCFLAGS=["-flto"])
             env.Append(LINKFLAGS=["-flto"])
+        # Workaround https://github.com/emscripten-core/emscripten/issues/19781.
+        cc_version = get_compiler_version(env)
+        cc_semver = (int(cc_version["major"]), int(cc_version["minor"]), int(cc_version["patch"]))
+        if cc_semver >= (3, 1, 42):
```

Comment:
I'll refactor this logic in #80719 after this is merged.

Reply:
Just noticed in the `3.x` branch that we use `tuple(get_compiler_version(env))` there, which is nicer. I'm changing it to that when cherry-picking, and for `master` I'll refactor later.